feat: 회원가입 및 로그인/로그아웃 구현

HaegyeongKim01 opened the pull request:
## 개요
<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->

Resolves: #10 


### ✅ 소셜 로그인(OAuth2)
- 카카오 기반 로그인 및 회원가입 처리 구현  
- OAuth2 인증 완료 후 JWT 토큰 발급 및 사용자 정보 저장  

---

### ✅ JWT 기반 인증 처리
- Spring Security + JWT 인증 필터 적용  
- Access / Refresh Token 생성 및 Redis 저장  
- HttpOnly 쿠키를 통한 토큰 관리  

---

### ✅ 로그아웃 기능 구현
- 서버 측 토큰 무효화 및 쿠키 제거  
- `/api/v1/member/logout` API 구현  

---

### ✅ 마이페이지 조회 API 구현
- 사용자 정보 조회를 위한 `/api/member/me` 엔드포인트 추가  
- `MemberService` 및 `MemberMapper` 연동  

## PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

HaegyeongKim01 committed:
Feat: 회원 소셜 로그인 구현

- 카카오 OAuth2.0 기반 로그인 및 회원가입 기능 추가
- Spring Security + JWT 기반 인증 처리 구현
- Refresh Token 발급 및 Redis 저장 로직 적용
- HttpOnly 쿠키를 통한 Access/Refresh 토큰 관리

HaegyeongKim01 committed:
Feat: 로그아웃 기능 구현

- 로그아웃 API 구현

HaegyeongKim01 committed:
Refactor: JwtAuthenticationFilter 리팩토링

- 인증 필터 리팩토링

HaegyeongKim01 committed:
Feat: MemberMapper 추가

- MapStruct로 MemberMapper 추가

HaegyeongKim01 committed:
Feat: 마이페이지 조회 API 추가

- GET /api/v1/member/me 엔드포인트 구현
- 로그인된 사용자 정보 기반 응답
- MemberService 및 MemberMapper 활용

HaegyeongKim01 committed:
Feat: 헤더바 로그인/로그아웃 상태에 따라 UI 구성

- Spring Security의 인증 상태 기반으로 헤더 영역에 동적 UI 구성
	- 로그인된 사용자에게는 '마이페이지' 및 '로그아웃' 버튼 노출
	- 비로그인 사용자에게는 '로그인' 버튼 노출
- Thymleaf의 sec:authorize 사용으로 인증 상태 처리

HaegyeongKim01 committed:
Fix: 로그아웃 기능 오류 수정

- 로그아웃 시 세션 종료 및 쿠키 삭제가 정상적으로 동작하지 않는 문제 수정

HaegyeongKim01 committed:
Fix: 로그아웃 후 access token 재발급 버그 수정

- Spring Security에 formLogin() 설정을 명시적으로 추가하여 인증 흐름 정상화
- .formLogin().loginPage("/login") 설정이 없을 경우, SecurityContext가 제대로 초기화 하지 않아,
    로그아웃 이후에도 access token이 계속 재생성되는 문제가 발생
- /login 페이지는 인증 필터 활성화를 위한 placeholder로 유지
- 이로 인해 OAuth2 로그인 후 쿠키 및 SecurityContext가 정상 동작하며, 로그아웃 후에도 토큰이 제거됨

HaegyeongKim01 committed:
Merge to dev

HaegyeongKim01 committed:
Refactor: Merge 이후 Member 관련 네이밍/타입 개선

- provider, role enum 변경
- 변수 네이밍 정리

HaegyeongKim01 committed:
Feat: 회원가입 이메일 정보 파싱 기능 추가

- 카카오 OAuth 응답에서 "kakao_account" 내부의 email 값을 파싱하도록 로직 수정

HaegyeongKim01 committed:
Refactor: 명시적 생성자 제거 및 Lombok 어노테이션 적용

- 명시적으로 작성된 생성자를 제거하고
  Lombok의 @RequiredArgsConstructor 어노테이션으로 대체
- 코드 간결성 향상 및 불필요한 보일러플레이트 제거

heodongwon commented on src/main/java/com/itcen/censquare/global/config/SecurityConfig.java:
```diff
@@ -0,0 +1,64 @@
+  }
+
+  @Bean
+  public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
```
CORS 설정은 필요없나요?

HaegyeongKim01 replied:
추후 도메인, 인증 방식을 고려하여 CORS 설정을 추가하겠습니다!!

heodongwon commented on src/main/java/com/itcen/censquare/domain/member/dto/MemberResDto.java:
```diff
@@ -0,0 +1,15 @@
+
+@Getter
+@RequiredArgsConstructor
+public class MemberResDto {
```
MemberRespDto
p 붙여주세욧

HaegyeongKim01 replied:
💬 넵! 오타가 있네요 바로 수정하겠습니다 감사합니다! 👍

HaegyeongKim01 committed:
Refactor: 클래스명 오타 수정 (MemberResDto → MemberRespDto)

- 클래스명 오타 수정 (Res → Resp)
- 관련된 import 및 참조도 함께 변경
- Response DTO 네이밍 컨벤션 통일 목적

heodongwon reviewed (approved):
LGTM
